src/mem/sizeclasstable.h

```diff
@@ -482,7 +482,7 @@ namespace snmalloc
     }
     if (size == 0)
     {
-      return 0;
+      return 1;
```

Comment:
Could you add a comment about returning `nullptr` versus minimum size allocation? And this line would need changing if we changed to `nullptr` return from `realloc`.  It is really useful to document this.

Also, I would prefer 
 ```suggestion
       return sizeclass_to_size(size_to_sizeclass(1));
```
which will compile to 
```
   return 16;
```
But it means we have
```
 malloc_usable_size(malloc(x, size)) == round_size(size)
```